crates/runtime/src/auth/mod.rs

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use crate::{
+    secrets::{ParamStr, Secrets},
+    App,
+};
+use runtime_auth::{api_key::ApiKeyAuth, FlightBasicAuth, GrpcAuth, HttpAuth};
+use secrecy::ExposeSecret;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+pub struct EndpointAuth {
+    pub http_auth: Option<Arc<dyn HttpAuth + Send + Sync>>,
+    pub flight_basic_auth: Option<Arc<dyn FlightBasicAuth + Send + Sync>>,
+    pub grpc_auth: Option<Arc<dyn GrpcAuth + Send + Sync>>,
```

Comment:
Is the api key based auth going to be implemented for flight & grpc as well?

Reply:
Yes, incremental PRs